contracts/WhitelistedCommitment.sol

```diff
@@ -89,35 +87,42 @@ contract WhitelistedCommitment is AccessControlled, AccessRoles, PublicCommitmen
             curve.burnNeumark(neumarks);
         }
     }
-    event Debug(uint256 value);
-    /// overrides base class to compute neumark reward for fixed price investors
+
+    /// overrides base class to compute neumark reward for ordered whitelistfix investors
     function giveNeumarks(address investor, uint256 eth, uint256 euros)
         internal
         returns (uint256)
     {
-        uint256 fixedTicket = fixedCost[investor]; // returns 0 in case of investor not in mapping
-        Debug(fixedTicket);
-        Debug(eth);
+        uint256 fixedRemainingTicket = fixedCostTickets[investor]; // returns 0 in case of investor not in mapping
+        uint256 fixedRemainingNeumarks = fixedCostNeumarks[investor]; // returns 0 in case of investor not in mapping
+
         // what is above limit for fixed price should be rewarded from curve
         uint256 reward = 0;
-        if ( eth > fixedTicket ) {
-            if (fixedTicket > 0) // recompute euro if part of msg.value goes thru whitelist
-                euros = convertToEUR(eth - fixedTicket);
-            reward = curve.issue(euros); // PublicCommitment.giveNeumarks(investor, eth - fixedTicket, euros);
-            eth = fixedTicket;
+        if ( eth > fixedRemainingTicket ) {
+            if (fixedRemainingTicket > 0) // recompute euro if part of msg.value goes thru whitelist
+                euros = convertToEUR(eth - fixedRemainingTicket);
+            reward = curve.issue(euros);
+            eth = fixedRemainingTicket;
         }
-        Debug(eth);
+
         // get pro rata neumark reward for any eth left
         uint256 fixedreward = 0;
         if (eth > 0) {
-            fixedreward = proportion(totalFixedCostNeumarks, eth, totalFixedCostAmount);
-            // rounding errors, send out remainders
-            // @remco review
-            uint256 remainingBalance = neumarkToken.balanceOf(address(this));
-            if (absDiff(fixedreward, remainingBalance) < 1000)
-                fixedreward = remainingBalance; // send all
-            // decrease ticket size
-            fixedCost[investor] -= eth;
+            fixedreward = proportion(fixedRemainingNeumarks, eth, fixedRemainingTicket);
+            // decrease ticket size and neumarks left
+            if ((fixedreward > fixedRemainingNeumarks)
+                || (fixedRemainingNeumarks - fixedreward < 10))
+            {
+                // give rest of the neumarks
+                fixedreward = fixedRemainingNeumarks;
+                // zero whole ticket
+                fixedCostNeumarks[investor] = 0;
+                fixedCostTickets[investor] = 0;
```

Comment:
This case is not necessary.

Reply:
This case can be removed if we use `min`.

Reply:
This case compares Neumarks not eth (I assume you refer to `min` usage below)
```
uint256 ticketEth = min(eth, fixedRemainingTicket);
```
We can remove this case if what you say of your `Disbursal` contract applies here: last proportion is surely 1/1 as all the rounding errors are annihilated before. Let me write some tests for it

Reply:
Ehhh ok it is trivial to prove that last proportion is 1:1 :) it does not matter at all how you round. you can even randomly round up or down